lib/cn/kvset.c

```diff
@@ -3776,37 +3776,63 @@ vgmap_free(struct vgmap *vgmap)
     free(vgmap);
 }
 
-/* TODO: Use binary search as an optimization */
-static merr_t
-vgmap_vbidx_out2adj(struct vgmap *vgmap, uint16_t vbidx_out, uint16_t *vbidx_adj)
+static uint32_t
+vgmap_bsearch_le(uint16_t *vbidx, uint32_t cnt, uint16_t key)
```

Comment:
Any particular reason why this is little endian instead of big endian? @nabeelmmd

Reply:
Heh..  Well, le should be faster on amd64, but prob not s390x.  I think we have a build define for desired endianess, rihgt? Probably we should consult that...

Reply:
he he.. sorry for the confusion, "le" expands to "less than or equals to". Probably I should use LE or LTE here?

Reply:
Maybe just remove `le` from the name, and document it in a comment above the function?

Reply:
if it's not an "eq" search, i feel like is shouldn't just be "vgmap_bsearch".

Reply:
Fixed.